app/components/form/fields/ComboboxField.tsx

```diff
@@ -69,25 +59,62 @@ export function ComboboxField<
     control,
     rules: { required, validate },
   })
-  const [selectedItemLabel, setSelectedItemLabel] = useState(
-    getSelectedLabelFromValue(items, field.value || '')
+
+  // Memoize description computation to avoid recalculating on every render
+  const memoizedDescription = useMemo(() => {
```

Comment:
I'd be surprised if this memo has a meaningful effect on performance

Reply:
Why would this trigger a re-render – since `description`, `placeholder` and `allowArbitraryValues` are likely the same values across the lifetime of this component. So the prop on `<Combobox>` wouldn't change.